src/app/errors/components/ErrorBoundary.tsx

```diff
@@ -0,0 +1,30 @@
+import React, { ReactNode } from 'react';
+import { connect } from 'react-redux';
+import Sentry from '../../../helpers/Sentry';
+import { Dispatch } from '../../types';
+import { recordError } from '../actions';
+
+interface Props {
+  children: ReactNode;
+  recordError: (error: Error) => void;
+}
+
+class ErrorBoundary extends React.Component<Props> {
+
+  public componentDidCatch(error: Error) {
+    this.props.recordError(error);
```

Comment:
recording the error in the state is nice because the pre-renderer needs a way to tell something went wrong, but the spec is to have the ErrorBoundary replace the content of the page with the error message, not pop the modal, so i think something around here has to change

Reply:
Yeah, that's mainly what I wanted your opinion on.  We could just render the error instead of the content, and in fact that was what I did originally.

The problem with that is that it'll only work if the boundary wraps a large enough area to display the error.  If the wrapped content is small or maybe does something funky like render a modal itself, then replacing children with the error will be unpredictable.

Perhaps the solution is to have `ErrorBoundary` take a `renderErrorAsModal` for cases like that, and otherwise render errors in place of children.

Thoughts?

Reply:
i think sort of the point is that we never want the `ErrorBoundary` to trigger a modal, because it represents the failure to render the content, we would just replace the content. we only want the modal for errors happening outside of the render process.

agree that if we used error boundaries on smaller elements, the design would get weird. what if we add a render prop for `errorDisplay` or something, and make a new component for `FullPageError` that we use in all the places you've currently got. that way if we wanted to later add a boundary around something like a sidebar or a note card, that would be easier to organize

Reply:
Hey, I think this got missed.  I've went through and left the error card only displaying inline for right now on the theory that we should only do as much as is required.

If and when we need to display errors in a small region we can do as you suggest then.